crates/ruff_cli/src/diagnostics.rs

```diff
@@ -241,96 +241,109 @@ pub(crate) fn lint_path(
             error: parse_error,
         },
         fixed,
-    ) = if matches!(fix_mode, flags::FixMode::Apply | flags::FixMode::Diff) {
-        if let Ok(FixerResult {
-            result,
-            transformed,
-            fixed,
-        }) = lint_fix(path, package, noqa, settings, &source_kind, source_type)
-        {
-            if !fixed.is_empty() {
-                match fix_mode {
-                    flags::FixMode::Apply => match transformed.as_ref() {
-                        SourceKind::Python(transformed) => {
-                            write(path, transformed.as_bytes())?;
-                        }
-                        SourceKind::IpyNotebook(notebook) => {
-                            let mut writer = BufWriter::new(File::create(path)?);
-                            notebook.write(&mut writer)?;
-                        }
-                    },
-                    flags::FixMode::Diff => {
-                        match transformed.as_ref() {
+    ) = match fix_mode {
+        flags::FixMode::Apply(fix_suggested) | flags::FixMode::Diff(fix_suggested) => {
+            if let Ok(FixerResult {
+                result,
+                transformed,
+                fixed,
+            }) = lint_fix(
+                path,
+                package,
+                noqa,
+                settings,
+                &source_kind,
+                source_type,
+                fix_suggested,
+            ) {
```

Comment:
nit: could you give this a temporary variable

Reply:
I'm not sure what you mean here, but I'd like to address this separately to keep the diff clear (I reverted some of Evan's changes here to keep the diff simpler already).